### PR TITLE
Remove deprecated annotations

### PR DESCRIPTION
**Pull request by SStorm**
## Summary of the changes / Why this is an improvement

There's been a significant amount of time and they are no longer used by anything.

This will require a 2.0 release as per SemVer.

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -29,6 +29,7 @@ Unreleased
 
 * Removed the Context class in favour of simple storing the context as a dictionary.
 
+* Removed the deprecated ``zalando...`` annotations. This will require a 2.0 release.
```
Comment (lukasbals) [inline CHANGES.rst:32]: I think it would be more consistent to add new changes at the top? 🤔

**Review by lukasbals — APPROVED** [review]

File: crate/operator/main.py [hunk 2]
```diff
@@ -123,33 +122,17 @@ async def startup(settings: kopf.OperatorSettings, **kwargs):
...
-        "The 'kopf.zalando.org/*' annotations and the "
-        "'kopf.zalando.org/KopfFinalizerMarker' finalizer are deprecated and will be "
-        "removed in version 2.0.",
-        DeprecationWarning,
-    )
-    # TODO: In version 2.0 change to:
-    # settings.persistence.diffbase_storage = kopf.AnnotationsDiffBaseStorage(
-    #     prefix=f"operator.{API_GROUP}", key="last", v1=False
-    # )
     settings.persistence.diffbase_storage = kopf.MultiDiffBaseStorage(
```
Comment (turbo-ele) [inline crate/operator/main.py:125]: is this still needed? (assumption from the TODO)

File: crate/operator/main.py [hunk 3]
```diff
@@ -123,33 +122,17 @@ async def startup(settings: kopf.OperatorSettings, **kwargs):
...
             ),
-            kopf.AnnotationsDiffBaseStorage(),  # For backwards compatibility
         ]
     )
     settings.persistence.finalizer = f"operator.{API_GROUP}/finalizer"
-    # TODO: In version 2.0 change to:
-    # settings.persistence.progress_storage = (
-    #     kopf.AnnotationsProgressStorage(prefix=f"operator.{API_GROUP}", v1=False),
-    # )
     settings.persistence.progress_storage = kopf.MultiProgressStorage(
```
Comment (turbo-ele) [inline crate/operator/main.py:133]: same here, i guess you can change it to the way as proposed in the todo

Comment (SStorm) [reply crate/operator/main.py:133]: Ah, thanks for this, I failed to notice that it was not MultiProgressStorage anymore.

File: CHANGES.rst (hunk 1 above)
Comment (SStorm) [reply CHANGES.rst]: TBH I was always confused by this and would usually add them at the bottom. Need to remember!... Changed now, thanks.

**Review by turbo-ele — APPROVED** [review]
👍

**Pull request merged**